### PR TITLE
infrastructure

### LINKED ISSUE #198: H3 test failure
Hey @knaaptime,

I don't think the failure we see in CI is coming from a different version of H3. In the last passing run, the version is `h3-py 3.7.6 py311hb755f60_1`. In the first failing it is exactly the same, meaning that the binary did not change.

The issue is caused by update of PROJ from 9.3.0 to 9.3.1. If I downgrade to 9.3.0, the test passes.

Windows is passing because this test is skipped on Windows.

My assumption is that the re-projection is now slightly different, likely more precise than before, and that we should just update the expected value.

### PR DESCRIPTION
**Pull request by knaaptime**
- modernize test workflow
- resolves #198 
- rm quilt3 as a test dependency

### CODE REVIEW

File: ci/311.yml [hunk 1]
```diff
@@ -20,11 +19,21 @@ dependencies:
...
   - astropy
+  - mapclassify
+  - descartes
+  - sphinx>=1.4.3
+  - sphinxcontrib-bibtex==1.0.0
+  - sphinx_bootstrap_theme
+  - numpydoc
+  - nbsphinx
+  - joblib
+  - astropy
```
Comment (jGaboardi) [inline ci/311.yml:39]: Actually need `astropy`? 😮

File: .github/workflows/unittests.yml [hunk 2]
```diff
@@ -8,52 +9,94 @@
...
-       - name: reinstall tobler - bash
-         shell: bash -l {0}
-         run: pip install -e . --no-deps --force-reinstall
-         if: matrix.os != 'windows-latest'
-      
-       - name: reinstall tobler - powershell
-         shell: powershell
-         run: pip install -e . --no-deps --force-reinstall
-         if: matrix.os == 'windows-latest'
+       ##############################################################################
```
Comment (jGaboardi) [inline .github/workflows/unittests.yml:57]: Do we need `pulp` for `tobler`?

File: ci/311.yml (hunk 1 above)
Comment (knaaptime) [reply ci/311.yml:39]: yeah, for pycno

File: ci/311.yml [hunk 3]
```diff
@@ -20,11 +19,21 @@ dependencies:
...
   - pytest-cov
+  - pytest-xdist
+  - coverage
   - twine
   - pip
   - h3-py
   - joblib
   - astropy
+  - mapclassify
+  - descartes
```
[the remark this answers is not in the document]
Comment (knaaptime) [reply ci/311.yml:32]: probably not

File: .github/workflows/unittests.yml (hunk 2 above)
Comment (knaaptime) [reply .github/workflows/unittests.yml:57]: no; its all commented out, but i'll remove it anyway

**Review by jGaboardi — APPROVED** [review]

**Pull request merged**